intake_xarray/base.py

```diff
@@ -1,38 +1,231 @@
+                da = da.assign_coords(**coords).chunk(self.chunks)
+            if isinstance(da, xr.DataArray):
+                da.name = data_var
+            merge_das.append(da)
+
+        return xr.merge(merge_das)
+
+    def _open_dataset(self):
+        from glob import glob
+
+        is_glob = '*' in self.urlpath
+        is_list = isinstance(self.urlpath, list)
+
+        # if the path doesn't need pattern reading and is multi - use multireader
+        if self.multireader and not (self.pattern or self.merge_dim) and is_glob:
+            self._ds = self.multireader(self.urlpath, chunks=self.chunks,
+                                        concat_dim=self.concat_dim, **self.kwargs)
+        elif is_glob:
+            files = sorted(glob(self.urlpath))
+            if len(files) == 0:
+                raise Exception("No files found at {}".format(self.urlpath))
+            self._ds = self._open_files(files)
+        elif is_list:
+            self._ds = self._open_files(self.urlpath)
+        else:
+            self._ds = self.reader(self.urlpath, chunks=self.chunks,
+                                   **self.kwargs)
 
     def _get_schema(self):
         """Make schema object, which embeds xarray object and some details"""
         from .xarray_container import serialize_zarr_ds
 
+        is_multi = '*' in self.urlpath or isinstance(self.urlpath, list)
         self.urlpath = self._get_cache(self.urlpath)[0]
+        if isinstance(self.urlpath, list) and not is_multi:
+            self.urlpath = self.urlpath[0]
 
         if self._ds is None:
             self._open_dataset()
 
+            if isinstance(self._ds, xr.DataArray):
+                da = self._ds
+                ds = xr.Dataset({'value': da})
+                extra_metadata = {'array': 'value'}
+            else:
+                ds = self._ds
+                da = ds[list(ds.data_vars)[0]]
+                extra_metadata = {}
+
             metadata = {
-                'dims': dict(self._ds.dims),
-                'data_vars': {k: list(self._ds[k].coords)
-                              for k in self._ds.data_vars.keys()},
-                'coords': tuple(self._ds.coords.keys()),
+                'dims': dict(ds.dims),
+                'data_vars': {k: list(ds[k].coords)
+                              for k in ds.data_vars.keys()},
+                'coords': tuple(ds.coords.keys()),
+                **extra_metadata
             }
             if getattr(self, 'on_server', False):
-                metadata['internal'] = serialize_zarr_ds(self._ds)
-            metadata.update(self._ds.attrs)
+                metadata['internal'] = serialize_zarr_ds(ds)
+
             self._schema = Schema(
                 datashape=None,
-                dtype=None,
-                shape=None,
-                npartitions=None,
+                dtype=str(da.dtype),
+                shape=da.shape,
+                npartitions=getattr(da.data, 'npartitions', None),
```

Comment:
This is populated only for the case of an array? I assume that, in general, the number of chunks can be different for each variable of a data-set.